### PR TITLE
docs: remove Dart edge docs

### PR DESCRIPTION
**Pull request by dshukertjr**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

- Removes the Dart Edge docs and redirects it to the Invertase's docs.
- Adds a redirect to the Invertase's docs for Dart Edge. 

## Additional Context

Users are getting confused about the state of Dart Edge, which hasn't been updated in a long time due to many breaking changes happening to Dart's development. 
https://x.com/illiaromanenko/status/1829522232155987979

### CODE REVIEW

File: apps/docs/next.config.mjs [hunk 1]
```diff
@@ -149,6 +149,14 @@ const nextConfig = {
         basePath: false,
         permanent: false,
       },
+
+      // Redirect Dart Edge docs to Invertase's docs
+      {
+        source: '/docs/guides/functions/dart-edge',
+        destination: 'https://invertase.docs.page/dart_edge/platform/supabase',
```
Comment (charislam) [inline apps/docs/next.config.mjs:156]: For future-forward security, I'm not a huge fan of redirecting to third parties unless they are super huge players that we expect to last forever (like AWS 😄), I'm happy enough to let this 404 or just redirect to functions index page

**Review by charislam — CHANGES_REQUESTED** [review]

**Comment by ivasilov** [comment]
@dshukertjr can you change the existing page to a short explanation and a link the user can click to go to the other docs?

File: apps/docs/next.config.mjs (hunk 1 above)
Comment (dshukertjr) [reply apps/docs/next.config.mjs:156]: Sorry, this PR slipped through the crack. Sounds good. Let's do that 👍

**Comment by dshukertjr** [comment]
@ivasilov Done!

**Comment by Tananga** [comment]
@dshukertjr  Is there any plans to add Dart support for Functions? This is a huge problem for Flutter devs

**Comment by dshukertjr** [comment]
@Tananga Not on our end, but possibly through collaboration like the Dart Edge project. I just wish the Dart team settles with the rapid breaking changes around WASM support so that a project like this could be sustained.

**Review by charislam — APPROVED** [review]

**Pull request merged**